library/java/src/io/envoyproxy/envoymobile/engine/AndroidEngine.java

```diff
@@ -2,22 +2,21 @@
 
 import android.content.Context;
 import android.net.ConnectivityManager;
-import io.envoyproxy.envoymobile.engine.types.EnvoyStatus;
 
 public class AndroidEngine {
 
   // Internal reference to helper object used to load and initialize the native library.
   // Volatile to ensure double-checked locking works correctly.
   private static volatile AndroidEngine loader = null;
 
-  private final Engine engine;
+  private final EnvoyEngine envoyEngine;
 
   // Private helper class used by the load method to ensure the native library and its
   // dependencies are loaded and initialized at most once.
   private AndroidEngine(Context context) {
     System.loadLibrary("envoy_jni");
```

Comment:
Why is this in the initializer? And wouldn't we want to load `envoy_jni` in the pure java case?

Reply:
Ok, sorry, I see the loader pattern below now. Should this be AndroidLoader instead? And there be a pure JavaLoader?

Reply:
Let's fix that in another PR and get the interfaces in?

Reply:
Sounds good.